src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -720,7 +720,7 @@ private void writeRecord(SinkRecord record) {
     if (!startOffsets.containsKey(encodedPartition)) {
       startOffsets.put(encodedPartition, record.kafkaOffset());
     }
-    offsets.put(encodedPartition, record.kafkaOffset());
```

Comment:
if we were adding the correct end offset on the file, I wonder how this bug was demonstrating itself actually. 

Seems that the exported files would have the right start and end offsets even for compacted topics. I quickly looked at how we use the variable `offset` which would be the one to get misaligned/miscomputed and seems that this would only influence the offset we commit to the consumers topic. But we only depend on this offset as a fall back and we still recover from the max offset across the files on HDFS (we didn't use to depend at all but this has become a fallback since some time now). Have you noticed something different?